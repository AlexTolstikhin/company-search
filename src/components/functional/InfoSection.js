import React from 'react';
import PropTypes from 'prop-types';


const styles = {
    infoSection: {
        borderRadius: '15px',
        fontSize: '15px',
        lineHeight: '20px',
        textAlign: 'justify',
        background: '#c7ede9',
        padding: '10px 20px 20px 20px',
        margin: '10px'
    }
}

const InfoSection = ({ text, title}) => 
    <div style={styles.infoSection}>
        <h2>{title}</h2>
        <hr />
        <div>{text}</div>
    </div>

InfoSection.defaultProps = {
    text: '',
    title: ''
};

InfoSection.propTypes = {
    text: PropTypes.string,
    title: PropTypes.string
};


export default InfoSection;