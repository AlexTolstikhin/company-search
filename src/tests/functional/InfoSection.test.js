import React from 'react';
import renderer from 'react-test-renderer';
import InfoSection from '../../components/functional/InfoSection';

it('snaps InfoSection with undefined props', () => {
	const tree = renderer
		.create(<InfoSection />)
		.toJSON();
	expect(tree).toMatchSnapshot();
});

it('snaps InfoSection with defined props', () => {
	const tree = renderer
		.create(<InfoSection title="foo" text="loo" />)
		.toJSON();
	expect(tree).toMatchSnapshot();
});