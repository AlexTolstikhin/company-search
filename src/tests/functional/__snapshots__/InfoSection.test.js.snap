// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snaps InfoSection with defined props 1`] = `
<div
  style={
    Object {
      "background": "#c7ede9",
      "borderRadius": "15px",
      "fontSize": "15px",
      "lineHeight": "20px",
      "margin": "10px",
      "padding": "10px 20px 20px 20px",
      "textAlign": "justify",
    }
  }
>
  <h2>
    foo
  </h2>
  <hr />
  <div>
    loo
  </div>
</div>
`;

exports[`snaps InfoSection with undefined props 1`] = `
<div
  style={
    Object {
      "background": "#c7ede9",
      "borderRadius": "15px",
      "fontSize": "15px",
      "lineHeight": "20px",
      "margin": "10px",
      "padding": "10px 20px 20px 20px",
      "textAlign": "justify",
    }
  }
>
  <h2>
    
  </h2>
  <hr />
  <div>
    
  </div>
</div>
`;
